python/ray/experimental/channel/common.py

```diff
@@ -333,6 +429,68 @@ def start(self):
         pass
 
     def _read_list(self, timeout: Optional[float] = None) -> List[Any]:
+        timeout = 1e6 if timeout is None or timeout == -1 else timeout
+        self._consume_non_complete_object_refs_if_needed(timeout)
+
+        (
+            waitables_to_num_consumers,
+            skip_deserialization_waitables_to_num_consumers,
+        ) = self._get_all_waitables_to_num_consumers()
+        normal_waitables = list(waitables_to_num_consumers.keys())
+        skip_deserialization_waitables = list(
+            skip_deserialization_waitables_to_num_consumers.keys()
+        )
```

Comment:
looks like these are static? should we do it at init time?

Reply:
move to ReaderInterface constructor.

Reply:
After giving it a second thought, I realized it is not static. For example, the `get_ray_waitables` method of `BufferedSharedMemoryChannel` should return the buffer that will be read in the current read operation. Therefore, the return value of `get_ray_waitables` is not always the same.

Reply:
```python
    def get_ray_waitables(self) -> List[Tuple[ObjectRef, bool]]:
        self.ensure_registered_as_reader()
        return self._buffers[self._next_read_index].get_ray_waitables()
```